tensorflow_quantum/python/differentiators/differentiator.py

```diff
@@ -179,6 +179,86 @@ def refresh(self):
             del self.expectation_op
         return self
 
+    @abc.abstractmethod
+    def get_gradient_circuits(self, programs, symbol_names, symbol_values):
+        """Transforms programs to allow gradient calculations on real QPUs.
+
+        In the classical setting, backpropagation can be used to compute
+        derivatives of an objective function f(x) with respect to its inputs x.
+        Backpropagation works by recursively applying the chain rule to the
+        graph of computations defining f.
+
+        Moving towards the quantum setting, suppose there is some objective
+        function f(|s(x)>), where |s(x)> is the state produced by some quantum
+        circuit with parameters x. If one simulates the circuit on a classical
+        computer, then one can similarly apply the chain rule to f(|s(x)>) to
+        obtain the derivative; this is the approach taken by the adjoint
+        differentiator in TFQ.
+
+        If the quantum state |s(x)> is computed on a quantum processing unit
+        (QPU), backpropagation is no longer possible. Instead, the classical
+        computer, where the objective function resides, can only access
+        bitstrings sampled from states produced on the QPU; derivatives must be
+        computed from these samples instead. This function exposes the circuits,
+        symbols, and values required for computing such gradients on real QPUs.
+
+        NOTE: most users should simply hand a differentiator to the circuit
+        executor of their choice (for example, see the docs for
+        `tfq.layers.Expectation`, which accepts a `differentiator` initializer).
+        This function is intended for advanced users who need more flexibility
+        than the standard workflow allows.
```

Comment:
This is kind of redundant. Having seen our taking gradient tutorial or being an expert on QC most people should already know this. What do we think of this description:

```python3
@abc.abstractmethod
def get_gradient_circuits(self, programs, symbol_names, symbol_values):
"""Return the circuits used to compute gradients for a given set of forward pass circuits.

Prepares (but does not execute) all intermediate circuits needed to calculate the gradients for the
given forward pass circuits specified by `programs`, `symbol_names` and `symbol_values`. The returned
`tf.Tensor` objects give all necessary information to recreate the internal logic of the differentiator with
finer grained control over the calculation process. <insert huge formula here>. This can be useful for advanced users when executing on QPU devices or defining custom differentiation methods.

Note: This feature is intended for advanced users who want more fine control during graph construction and
execution. 

"""
```

Reply:
I like this, describes what the function more efficiently.